arkouda/groupbyclass.py

```diff
@@ -742,7 +763,27 @@ def argmax(self, values: pdarray) -> Tuple[groupable, pdarray]:
         (array([2, 3, 4]), array([9, 3, 2]))
         """
 
-        return self.aggregate(values, "argmax")
+        k, v = self.aggregate(values, "argmax")
+        return k, cast(pdarray, v)
+
+    def _nested_grouping_helper(self, values: groupable) -> groupable:
+        unique_key_idx = self.broadcast(arange(self.ngroups), permute=True)
+        if hasattr(values, "_get_grouping_keys"):
+            # All single-array groupable types must have a _get_grouping_keys method
+            if isinstance(values, pdarray) and cast(pdarray, values).dtype == akfloat64:
+                raise TypeError("grouping/uniquing unsupported for float64 arrays")
+            togroup = [unique_key_idx, values]
+        else:
+            # Treat as a sequence of groupable arrays
+            for v in values:
+                if (
+                    isinstance(values, pdarray)
+                    and cast(pdarray, values).dtype != akint64
+                    and cast(pdarray, values).dtype != akuint64
+                ):
```

Comment:
two things here
- we can apply the same short circuiting logic to avoid an explicit cast
- `v` never seems to be used here. I think `values` is supposed to `v` here

I think we should change this to 
```python
for v in values:
    if isinstance(v, pdarray) and v.dtype not in [akint64, akuint64]:
```